spec/grape/validations/params_scope_spec.rb

```diff
@@ -155,6 +155,16 @@ def initialize(value)
       expect(last_response.status).to eq(400)
       expect(last_response.body).to eq('foo is empty')
     end
+
+    it do
+      subject.params do
+        requires :foo, as: :bar, allow_blank: false
+      end
+      subject.get('/alias-not-blank-with-value') {}
+      get '/alias-not-blank-with-value', foo: 'any'
+
+      expect(last_response.status).to eq(200)
```

Comment:
I get status 400 here with the `last_response.body` = `foo is empty`.

As the alias (`as`) is called before the `allow_blank`, it deletes the `foo` key from `params` and sets `params[:bar]` with its value.

Given that alias is just for internal usage (inside `get` block I mean), I think this should not fail, cause the `foo` param was given as expected. Do you agree?

Reply:
I'd tend to agree that this looks like a bug to me. Could you open an issue for this?

Reply:
(Thanks for looking at this!)